pkg/metricsadapter/controller.go

```diff
@@ -181,18 +191,23 @@ func (m *MetricsController) handleClusters() bool {
 			klog.Warningf("unable to access cluster %s, Error: %+v", clusterName, err)
 			return true
 		}
+		_ = m.TypedInformerManager.ForCluster(clusterName, clusterClient.KubeClient, 0)
 		_ = m.InformerManager.ForCluster(clusterName, clusterDynamicClient.DynamicClientSet, 0)
 	}
 	err = m.MultiClusterDiscovery.Set(clusterName)
 	if err != nil {
 		klog.Warningf("failed to build discoveryClient for cluster(%s), Error: %+v", clusterName, err)
 		return true
 	}
-	sci := m.InformerManager.GetSingleClusterManager(clusterName)
+
+	typedSci := m.TypedInformerManager.GetSingleClusterManager(clusterName)
 	// Just trigger the informer to work
-	_ = sci.Lister(provider.PodsGVR)
-	_ = sci.Lister(provider.NodesGVR)
+	_, _ = typedSci.Lister(provider.PodsGVR)
+	_, _ = typedSci.Lister(provider.NodesGVR)
 
+	typedSci.Start()
+	_ = typedSci.WaitForCacheSync()
+	sci := m.InformerManager.GetSingleClusterManager(clusterName)
```

Comment:
so, is this `SingleClusterManager` of `InformerManager` still necessary to keep it?

Reply:
yes.

Because metrics and custom-metrics still need